meetings/2018-01-17/minutes.md

```diff
@@ -0,0 +1,417 @@
+implementation. He noted that Bloop may be particularly useful for IntelliJ,
+though is uncertain about whether it could easily replace IntelliJ's own
+typechecker.
+
+### ScalaIDE/Eclipse - Iulian Dragos, Triplequotes
+
+Iulian has been working on the LSP implementation for VSCode, based on Ensime.
+The implementation delegates the protocol as an in-process Ensime instance,
+circumventing the socket protocol.
+
+This approach avoided many of the problems Iulian faced working on the ScalaIDE
+for Eclipse (which he spent approximately four years on). The ScalaIDE used the
+Scalac presentation compiler directly, through an asynchronous API that was
+able to interact directly with the typechecker. It worked happily with
+incomplete on incorrect programs, providing errors as the user typed, and
+offering early feedback.
+
+All ASTs were available to the IDE, and offered advanced, precise and rich
+information. Unfortunately there was a tight coupling between the plugin and
+the compiler. Memory requirements were high, because each different classpath
+required a different instance of the compiler, without sharing of (for example)
+symbol tables. Iulian is very much in favor of the language-server approach as
+it avoids many of these problems, though having a presentation compiler also
+provides many benefits, particularly for providing feedback for incomplete or
+non-compiling programs.
+
+### Dotty IDE/VSCode - Guillaume Martres, EPFL
+
+Any Dotty project built with sbt can launch VSCode with a single command, and
+will automatically provide common IDE features like code-linking and
+completion. Guillaume suggests that he would prefer the IDE to interact
+directly with the language server, rather than having the IDE interact with the
+build tool (which then interacts with the language server), and that this
+approach has been accommodated in the [Build Server Protocol
+proposal](https://github.com/scalacenter/bsp/blob/master/docs/bsp.md) developed
+by Jorge and Olaf.
+
+Guillaume also spoke about syntax highlighting, and debugging support in IDEs.
+
+### Dotty and TASTY - Martin Odersky, EPFL
+
+Martin had to leave the call early, so he talked about the Dotty compiler
+slightly earlier than scheduled.
+
+Dotty has a presentation compiler, which he considers important for providing
+information in the case of errors, but the editor also needs support for
+building. Currently sbt has to be spun up in a separate terminal. He proposes a
+protocol where *n* editors can talk to *m* build tools, but would start with
+just a few.
+
+Martin suggests that the presentation compiler could have configuration to
+start up any build tool, possibly with an overridable favourite, and then talk
+to that build tool through a common BSP, which would share the classpath and
+other configuration.
+
+He proposes starting with sbt, which already has an LSP implementation, but
+inviting other build tools such as Bloop to use the proposal.
```

Comment:
Can we mention here that bloop already implements bsp? Regarding sbt, you may want to mention that the Scala Center is collaborating with the sbt team at Lightbend to implement a bsp prototype and refine the specification.

Reply:
“A bsp server prototype in sbt” if it’s nit sufficiently clear